### PR TITLE
Fetosense API - variable name change and generate PDF correction

### PR DESCRIPTION
**Pull request by devikasuresh20**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.  

# Type of change  

- [x] Bug fix  
- [ ] New feature  
- [ ] Enhancement  
- [ ] Refactoring  
- [ ] Documentation  
- [ ] Other ( please specify )  

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

# Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by devikasuresh20** [comment]
@drtechie and @ravishanigarapu 
As you can see there is a codeql error in FetosenseServiceImpl-> generatePDF method. As a fix , I given condition to check the protocol in the reportPath for sanitization. But still codeql error is coming.
Can we make this as a false positive case ? Pls suggest on this.

**Comment by drtechie** [comment]
@devikasuresh20 As far as I understand, CodeQL has detected `reportPath`as a user provided input.
If you check the paths for this error, it looks like reportPath is coming via an API.
This URL is in turn used to initiate a connection - the reason for this security error.
I am not sure of the context, if the report is in an external system and if you know the domain of the `reportPath`, then add that as a criterion (see the example code provided in the issue).

**Comment by devikasuresh20** [comment]
@drtechie 
I added condition to check the domain of reportPath , but still same issue. Here I am comparing the domain using "contains()", because reportPath is different for each record.

**Comment by devikasuresh20** [comment]
@drtechie  As discussed, I added host name check also. But still same error from codeql. 
In HWC API, the same logic is working fine without any domain check.

@drtechie  and @ravishanigarapu  Can we make it as false positive ?

File: src/main/java/com/iemr/tm/service/foetalmonitor/FoetalMonitorServiceImpl.java [hunk 1]
```diff
@@ -176,40 +178,70 @@ public int updateFoetalMonitorData(FoetalMonitor foetalMonitorDataOutside) throw
...
 		} finally {
-			if (con != null) {
-				con.disconnect(); // Close the HTTP connection in the finally block
-			}
+			con.disconnect();
 		}
 
+		return filePathLocal;
 	}
+
```
Comment (drtechie) [inline src/main/java/com/iemr/tm/service/foetalmonitor/FoetalMonitorServiceImpl.java:210]: Can we remove unused code?

**Comment by drtechie** [comment]
@devikasuresh20 sounds okay to me.

File: src/main/java/com/iemr/tm/service/foetalmonitor/FoetalMonitorServiceImpl.java (hunk 1 above)
Comment (devikasuresh20) [reply src/main/java/com/iemr/tm/service/foetalmonitor/FoetalMonitorServiceImpl.java:210]: yes

**Review by ravishanigarapu — APPROVED** [review]
looks ok to me

**Pull request merged**